File: Data Structures & Algorithms/max-area-of-island/submission-0.java
class Solution {

    int[][] directions = new int[][]{{1,0},{-1,0},{0,1},{0,-1}};

    public int maxAreaOfIsland(int[][] grid) {
        int[] curArea=new int[]{0};
        int maxArea=0;
    
        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(grid[i][j]==1){
                    dfs(grid,i,j,curArea);
                    if(curArea[0]>maxArea) maxArea = curArea[0];
                    curArea[0]=0;
                }
            }
        }
        return maxArea;
    }

    void dfs(int[][] grid, int i,int j,int[] curArea){
        if(i<0 || j<0|| i>=grid.length || j>=grid[0].length|| grid[i][j]==0){
            return;
        }
        grid[i][j] = 0;
        curArea[0]++;
        
        for(int m=0;m<directions.length;m++)
            dfs(grid,i+directions[m][0],j+directions[m][1],curArea);

    }
}

File: Data Structures & Algorithms/surrounded-regions/submission-0.py
class Solution:
    def solve(self, board: List[List[str]]) -> None:
        ROWS = len(board)
        COLS = len(board[0])
        connZeroes = set()
        directions = [[1,0],[-1,0],[0,1],[0,-1]]

        def dfs(board,i,j):

            if i<0 or i==ROWS or j<0 or j==COLS or board[i][j]!='O' or ((i,j) in connZeroes):
                return
            
            connZeroes.add((i,j))
            for dir in directions:
                dfs(board,i+dir[0],j+dir[1])


        for i in range(ROWS):
            if board[i][0]=='O':
                dfs(board,i,0)
            if board[i][COLS-1]=='O':
                dfs(board,i,COLS-1)

        for i in range(COLS):
            if board[0][i]=='O':
                dfs(board,0,i)
            if board[ROWS-1][i]=='O':
                dfs(board,ROWS-1,i)
        print(connZeroes)
        
        for i in range(ROWS):
            for j in range(COLS):
                if (i,j) not in connZeroes and board[i][j]=='O':
                    board[i][j]='X'
        
